dixTable.py

```diff
@@ -229,41 +228,24 @@ def updateOrCreatePages(location, pairName, updateStats, editToken):
     languages = set(comprehensiveList(args.languages))
     logging.info('Using languages {}'.format(languages))
 
-    dirs = [
-        ('incubator', r'<name>(apertium-(\w{2,3})-(\w{2,3}))</name>'),
-        ('nursery', r'<name>(apertium-(\w{2,3})-(\w{2,3}))</name>'),
-        ('staging', r'<name>(apertium-(\w{2,3})-(\w{2,3}))</name>'),
-        ('trunk', r'<name>(apertium-(\w{2,3})-(\w{2,3}))</name>'),
+    dirPaths = [
+        'incubator',
+        'nursery',
+        'staging',
+        'trunk',
     ]
 
-    for (dirPath, dirRegex) in dirs:
-        svnData = str(subprocess.check_output('svn list --xml https://svn.code.sf.net/p/apertium/svn/%s/' % dirPath, stderr=subprocess.STDOUT, shell=True), 'utf-8')
-        for langDir in re.findall(dirRegex, svnData, re.DOTALL):
+    gitData = json.loads(urllib.request.urlopen("https://apertium.projectjj.com/stats-service/packages/").read())["packages"]
+    for package in gitData:
+        langDir = re.findall(r"(apertium-(\w+)-(\w+))", package["name"], re.DOTALL)
+        if langDir:
+            langDir = langDir[0]
+            dirPath = ''.join(set(package["topics"]) & set(["apertium-%s" % dir for dir in dirPaths]))[len("apertium-"):]
```

Comment:
Is there a reason that `dirPaths` can't just be constructed with the value of `set(["apertium-%s" % dir for dir in dirPaths]`?

Reply:
I’m not exactly sure what you mean by this, but:

- If it’s why it needs need the `package[“topics”]` it’s because it contains the correct `dirPath` that I need to find out
- If it’s why I take out `apertium-`, it’s because what the code checks for later on is just the bit after `apertium-`

- If it’s why I use `‘’.join()`, it’s because it creates a set and I just want a string

Reply:
> If it’s why I use ''.join(), it’s because the code creates a single element set and I want it as a string

I think this would be much clearer as just `list(...)[0]`.

> I’m not exactly sure what you mean by this, but:

`dirPaths` can start as `{'apertium-incubator', ...}`

Reply:
> `dirPaths` can start as `{'apertium-incubator', ...}`

Ah yes, yes it can.